Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -652,6 +658,43 @@ def test_connection(self):
 ''' HELPER FUNCTIONS '''
 
 
+def safely_update_context_data(func: Callable):
+    """Decorator for updating context data using versions.
+    In case of a race condition, preform func with the new context_data and try updating again.
+
+    Args:
+        func: The function to preform with the new context data before updating.
+
+    raise ValueError if context_data or version are not in the kwargs for the function.
+    raise DemistoException if reached maximum of retries.
+    """
+    def wrapper(*args, **kwargs):
+        context_was_set = False
+        retries = 0
+        max_retries = 5
+        return_value = None
+        while not context_was_set and retries < max_retries:
+            context_data, version, return_value = func(*args, **kwargs)
+            print_debug_msg(f'Attempting to update context data after version {version} with retry {retries}')
+            new_context_data, new_version = get_integration_context_with_version()
+            if new_version == version:
+                set_to_integration_context_with_retries(context_data)
```

Comment:
There could be a race condition.
Scenario of where between the call to get_integration_context_with_version 2 lines above until here a new version was set.

This can be solved by adding max_retry_times=1 to set_to_integration_context_with_retries and wrapping it with try except. WDYT? or am i wrong and no race conditions can occur